server/src/main/java/ru/practicum/shareit/exception/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,71 @@
+package ru.practicum.shareit.exception;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.MissingRequestHeaderException;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RestControllerAdvice
+public class GlobalExceptionHandler {
+    private static final Logger logger = LoggerFactory.getLogger(GlobalExceptionHandler.class);
+
+    @ExceptionHandler(NotFoundException.class)
+    @ResponseStatus(HttpStatus.NOT_FOUND)
+    public ErrorResponse handleNotFoundException(NotFoundException notFound) {
+        logger.error("Error: {}", notFound.getMessage());
+        return new ErrorResponse(HttpStatus.NOT_FOUND.value(), "Not Found", notFound.getMessage());
+    }
+
+    @ExceptionHandler(ValidationException.class)
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    public ErrorResponse handleValidationException(ValidationException validationException) {
+        logger.error("Error: {}", validationException.getMessage());
+        return new ErrorResponse(HttpStatus.BAD_REQUEST.value(), "Bad request", validationException.getMessage());
+    }
+
+    @ExceptionHandler(ConflictException.class)
+    @ResponseStatus(HttpStatus.CONFLICT)
+    public ErrorResponse handleConflictException(ConflictException ex) {
+        return new ErrorResponse(HttpStatus.CONFLICT.value(), "Conflict error", ex.getMessage());
+    }
+
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    public ErrorResponse handleValidationException(MethodArgumentNotValidException exception) {
+        List<String> errorMessages = exception.getBindingResult().getFieldErrors().stream().map(error -> String.format("Field '%s': %s", error.getField(), error.getDefaultMessage())).collect(Collectors.toList());
```

Comment:
Стримы принято бить по операциям
```suggestion
        List<String> errorMessages = exception.getBindingResult().getFieldErrors().stream()
            .map(error -> String.format("Field '%s': %s", error.getField(), error.getDefaultMessage()))
            .collect(Collectors.toList());
```